.test-infra/jenkins/Infrastructure.groovy

```diff
@@ -108,10 +108,13 @@ class Infrastructure {
 
   static void scaleCluster(def context, String jobName, Integer workerCount) {
     context.steps {
-      // Keep one extra Dataproc VM for Flink's Job Manager
-      workerCount += 1
-      shell("echo Changing number of workers to ${workerCount}")
-      shell("gcloud dataproc clusters update ${getClusterName(jobName)} --num-workers=${workerCount} --quiet")
+      environmentVariables {
+        env("FLINK_NUM_WORKERS", workerCount)
+      }
+      shell("gcloud dataproc clusters delete ${getClusterName(jobName)} --quiet")
```

Comment:
Unfortunately, I've found no better solution than recreating the cluster with a new number of workers. We can get back to this as soon as we find another way of scaling down the cluster.

This code here is repetitive but it will be fixed in the ongoing refactor.

Reply:
Can we at least reuse `teardownDataproc` and `setupFlinkCluster` here? Is it enough to just call the script? I'm afraid that when we run the script only, not all required env variables will be set (am I right?)

Reply:
I'm afraid reusing `teardownDataproc` is not possible because it appends cluster deleting to postBuildScripts. `setupFlinkCluster` on the other hand is troublesome (due to large number of arguments) and redundant (env variables are preserved because `context` is the same entity).

Reply:
I see. Ok. In this case let's leave as it is. I am aware (and reviewing) your refactoring effort in this class so we will clean it up there. 👍